### PR TITLE
Bug 1815253 - Record what caused an RKV db load failure

### PR DESCRIPTION
**Pull request by perrymcmanis144**
WIP for instrumenting which RKV failure caused a new rkv to be created

### CODE REVIEW

**Comment by perrymcmanis144** [comment]
@badboy alright probably ready for a look just to make sure it matches your expectations while I work through test failures.

File: glean-core/metrics.yaml [hunk 1]
```diff
@@ -706,6 +706,20 @@ glean.database:
...
     expires: never
 
+  rkv_load_result:
+    type: string
+    description: |
+      The result of attempting to load the RKV database.
+    bugs:
+      - https://bugzilla.mozilla.org/show_bug.cgi?id=1815253
+    data_reviews:
+      - https://bugzilla.mozilla.org/show_bug.cgi?id=todo
```
Comment (badboy) [inline glean-core/metrics.yaml:716]: Needs an update here.

File: glean-core/metrics.yaml [hunk 2]
```diff
@@ -706,6 +706,20 @@ glean.database:
       - glean-team@mozilla.com
     expires: never
 
+  rkv_load_result:
+    type: string
+    description: |
+      The result of attempting to load the RKV database.
```
Comment (badboy) [inline glean-core/metrics.yaml:712]: Should we mention here more precisely that it's only the errors and that it is not reported on successful loads?

File: glean-core/src/internal_metrics.rs [hunk 3]
```diff
@@ -277,6 +280,15 @@ impl DatabaseMetrics {
                 },
                 MemoryUnit::Byte,
             ),
+
+            rkv_load_result: StringMetric::new(CommonMetricData {
+                name: "rkv_load_result".into(),
+                category: "glean.error".into(),
+                send_in_pings: vec!["metrics".into()],
+                lifetime: Lifetime::Application,
```
Comment (badboy) [inline glean-core/src/internal_metrics.rs:288]: The metrics.yaml doesn't set a lifetime, so it defaults to `ping`, which I think makes more sense? We don't want this to be sent in every subsequent metrics ping until a restart, do we?

**Review by badboy — CHANGES_REQUESTED** [review]

**Review by badboy — APPROVED** [review]

**Pull request merged**